File: java/hamming/src/main/java/Hamming.java
public class Hamming {

  private String string1;
  private String string2;

  public Hamming(String string1, String string2){
    this.string1 = string1;
    this.string2 = string2;
    if(string1.length() != string2.length()){
      throw new IllegalArgumentException();
    }

  }

  public int getHammingDistance(){
        int count = 0;
    int letter_index = 0;
    for(int i = 0; i < string1.length(); i++){
      if(string1.charAt(letter_index) != string2.charAt(letter_index)){
        letter_index ++;
        count ++;
      }
      else{
        letter_index ++;
      }
    }
    return count;

  }
}

File: ruby/hamming/hamming.rb
class Hamming

    def self.compute(string1, string2)
      unless string1.length == string2.length
        raise ArgumentError
      end        
        count = 0
        letter_index = 0

          for i in 0..string1.length
          
          if string1[letter_index] != string2[letter_index]
            letter_index += 1
            count += 1

          else
            letter_index += 1
            
          end
      
    end
    return count
  end
end

module BookKeeping
  VERSION = 3 # Where the version number matches the one in the test.
end